staging/src/k8s.io/client-go/tools/remotecommand/remotecommand_test.go

```diff
@@ -264,3 +297,75 @@ func v4WriteStatusFunc(stream io.Writer) func(status *apierrors.StatusError) err
 		return err
 	}
 }
+
+// A writer wrap which provides a helper method to block until written.
+type writerWrap struct {
+	written chan bool
+	closed  bool
+	io.Writer
+}
+
+func newWriterWrap(w io.Writer) *writerWrap {
+	return &writerWrap{
+		written: make(chan bool),
+		Writer:  w,
+	}
+}
+
+func (w *writerWrap) BlockUntilWritten() {
+	<-w.written
+}
+
+func (w *writerWrap) Write(p []byte) (n int, err error) {
+	if !w.closed {
+		w.written <- true
+		close(w.written)
+		w.closed = true
+	}
+	return w.Writer.Write(p)
+}
+
+// `Executor.StreamWithContext` starts a goroutine in the background to do the streaming
+// and expects the deferred close of the connection leads to the exit of it on cancellation.
```

Comment:
How about  "... lead to the exit of the goroutine on cancellation" to be a little more clear?

Reply:
Edited.